OloEngine/tests/Rendering/RenderingRegressionTest.cpp

```
@@ -0,0 +1,86 @@
+#include "OloEngine/Scene/Components.h"
+
+#include <cstring>
+
+using namespace OloEngine; // NOLINT(google-build-using-namespace) — test file, brevity preferred
+
+// =============================================================================
+// ShadowUBO — Regression: all i32 fields must default to 0
+// Prevents accidental cascade-debug-visualization or wrong shadow state.
+// =============================================================================
+
+TEST(RenderingRegression, ShadowUBOIntFieldsDefaultZeroed)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)